torch/jit/__init__.py

```diff
@@ -1224,30 +1224,12 @@ def forward(self, input):
         _check_directly_compile_overloaded(obj)
         ast = get_jit_def(obj)
         if _rcb is None:
-            _rcb = _gen_rcb(obj, _frames_up)
+            _rcb = _jit_internal.createResolutionCallbackFromClosure(obj)
         fn = torch._C._jit_script_compile(qualified_name, ast, _rcb, get_default_args(obj))
         # Forward docstrings
         fn.__doc__ = obj.__doc__
         return fn
 
-def _gen_rcb(obj, _frames_up):
-    _frames_up = _frames_up + 1  # for invoking _gen_rcb()
-
-    closure_rcb = _jit_internal.createResolutionCallbackFromClosure(obj)
-    stack_rcb = _jit_internal.createResolutionCallback(_frames_up + 1)
```

Comment:
It's worth examining the remaining uses of `createResolutionCallback` to check if they are being used properly. For example, the one on line 1220 on this file seems suspicious if we are trying to recursively script classes (which we're not doing today, I think?).

Reply:
I went through and renamed getResolutionCallback to getResolutionCallbackFromFrame. The remaining use cases are correct and are (1) used in `define`, where we want to take the callers scope and resolve variables in it, and (2) for  class/interface declarations which are always annotated. There is a recursive class compilation case that I noticed that David added, but it isn't applied consistently yet, so I am not going to touch it yet.